src/middlewared/middlewared/main.py

```diff
@@ -850,19 +850,23 @@ def add_schema(self, schema):
     def get_schema(self, name):
         return self.__schemas.get(name)
 
-    async def run_in_thread_pool(self, pool, method, *args, **kwargs):
+    async def run_in_executor(self, pool, method, *args, **kwargs):
         """
-        Runs method in a native thread using concurrent.futures.ThreadPool.
-        This prevents a CPU intensive or non-greenlet friendly method
+        Runs method in a native thread using concurrent.futures.Pool.
+        This prevents a CPU intensive or non-asyncio friendly method
         to block the event loop indefinitely.
+        Also used to run non thread safe libraries (using a ProcessPool)
         """
         loop = asyncio.get_event_loop()
         task = loop.run_in_executor(pool, functools.partial(method, *args, **kwargs))
         await task
         return task.result()
 
     async def run_in_thread(self, method, *args, **kwargs):
-        return await self.run_in_thread_pool(self.__threadpool, method, *args, **kwargs)
+        return await self.run_in_executor(self.__threadpool, method, *args, **kwargs)
+
+    async def run_in_proc(self, method, *args, **kwargs):
+        return await self.run_in_executor(self.__procpool, method, *args, **kwargs)
 
     async def run_in_io_thread(self, method, *args):
```

Comment:
`run_in_io_thread` should also use `run_in_executor` then

Reply:
The time did I did it there was no run_in_io_thread and forgot to change after rebase, will fix.